File: sbur-redis/src/main/java/com/example/sburredis/PlaneFinderPoller.java
package com.example.sburredis;

import org.springframework.data.redis.connection.RedisConnectionFactory;
import org.springframework.data.redis.core.RedisOperations;
import org.springframework.scheduling.annotation.EnableScheduling;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;
import org.springframework.web.reactive.function.client.WebClient;

@EnableScheduling
@Component
public class PlaneFinderPoller {
    private WebClient client = WebClient.create("http://localhost:7634/aircraft");
    private final AircraftRepository aircraftRepository;
    private final RedisConnectionFactory factory;

    public PlaneFinderPoller(AircraftRepository aircraftRepository, RedisConnectionFactory factory) {
        this.aircraftRepository = aircraftRepository;
        this.factory = factory;
    }

    @Scheduled(fixedRate = 1000)
    private void pollPlanes() {
//        factory.getConnection().serverCommands().flushDb();

        client.get()
                .retrieve()
                .bodyToFlux(AirCraft.class)
                .filter(airCraft -> !airCraft.getReg().isEmpty())
                .toStream()
                .forEach(aircraftRepository::save);

        aircraftRepository.findAll().forEach(System.out::println);
    }
}

File: mongo-app/src/main/kotlin/com/example/mongoapp/PlaneFinderPoller.kt
package com.example.mongoapp

import org.springframework.scheduling.annotation.EnableScheduling
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Component
import org.springframework.web.reactive.function.client.WebClient
import org.springframework.web.reactive.function.client.bodyToFlux

@Component
@EnableScheduling
class PlaneFinderPoller(private val repository: AircraftRepository) {
    private val client: WebClient = WebClient.create("http://localhost:7634/aircraft");

    @Scheduled(fixedRate = 1000)
    private fun poll() {
        repository.deleteAll();

        client.get().retrieve()
            .bodyToFlux<Aircraft>()
            .filter { !it.reg.isNullOrEmpty() }
            .toStream()
            .forEach(repository::save)

        repository.findAll().forEach{ println(it)}
    }
}